### ISSUE TITLE
Code not working on VCM

### ISSUE THREAD

**Issue by EricaSkerrett** [EricaSkerrett/bme590final#99]
`(venv) vcm@vcm-7311:~/bme590final$ python3 final.py`
`/home/vcm/bme590final/venv/lib/python3.5/site-packages/skimage/viewer/__init__.py:6: UserWarning: Viewer requires Qt`
  `warn('Viewer requires Qt')`
 `* Serving Flask app "final" (lazy loading)`
 `* Environment: production`
   `WARNING: Do not use the development server in a production environment.`
   `Use a production WSGI server instead.`
 `* Debug mode: off`

I have the server, `final.py` running on a vcm on screen. When we run the server locally, it runs fine. But when we switch to the vcm address, we get KeyErrors when we run the gui:

`$ python gui.py`
`python gui.py`
`C:/Users/esker/MedicalDeviceSoftware/bme590final/capy.jpg`
`C:/Users/esker/MedicalDeviceSoftware/bme590final/capy.jpg`
`testtest@duke.edu`
`ReverseVideo`
`capy`
`dict_keys([])`
`Traceback (most recent call last):`
  `File "gui.py", line 366, in reverse`
    `self.next = App5()`
  `File "gui.py", line 388, in __init__`
    `self.init_gui()`
  `File "gui.py", line 398, in init_gui`
    `self.display_images()`
  `File "gui.py", line 416, in display_images`
    `decode(processed_images[image_name]))`
`KeyError: 'capy'`
`(venv)`

Everything is up to date on the vcm and requirements.txt is installed. Could it be because of the 'Viewer requires QT` line? Do you have any advice on how to troubleshoot this or what may be wrong? Thank you! @suyashkumar @mlp6 @dward2

**Comment by sputney13** [comment]
To clarify, when we run our gui on the vcm, the reason why we get the key error is because the image is not POSTing to the database. When I run the gui locally, I get the following (different) error, which is expected at the moment, and the processed image successfully posts to the database.

<img width="503" alt="screenshot" src="https://user-images.githubusercontent.com/42679463/49955829-cf0a9f80-fed1-11e8-9dcb-0eb16b143fe6.png">

**Comment by dward2** [comment]
Sorry, but I cannot seem to recreate your error, nor the same console output.  Can you create an test_function that I can run that generates the error?  Is the code you are running above on the master branch?   I am getting the same results whether I run your server locally or on a VM.  I start up the server (`final.py`) and then start the GUI (`gui.py`).  In the GUI, it appears that I am able to set up a new user (or access an existing user), choose an image, click on Upload, and then select HistogramEqualization.  The GUI window closes, and I see the following in my python console:
```
david.a.ward@duke.edu

HistogramEqualization
posted



Process finished with exit code -1073740791 (0xC0000409)
```
I don't see anything printed in the server console, other than the usual server start-up text.  The `final.log` file does contain this:
```
ERROR:root:User not in the database.
NoneType: None
ERROR:root:User not in the database.
NoneType: None
ERROR:root:User not in the database.
NoneType: None
ERROR:root:User not in the database.
NoneType: None
ERROR:root:User not in the database.
NoneType: None
ERROR:root:User not in the database.
NoneType: None
ERROR:root:User not in the database.
NoneType: None
INFO:root:Return user accessed.
INFO:werkzeug:67.159.89.155 - - [13/Dec/2018 13:35:15] "GET /image/user/david.a.ward@duke.edu HTTP/1.1" 200 -
INFO:werkzeug:67.159.89.155 - - [13/Dec/2018 13:35:54] "POST /image/upload/david.a.ward@duke.edu HTTP/1.1" 200 -
ERROR:flask.app:Exception on /image/processed/upload [POST]
Traceback (most recent call last):
  File "C:\Users\daw74\PycharmProjects\EricaFinal\venv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\daw74\PycharmProjects\EricaFinal\venv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\daw74\PycharmProjects\EricaFinal\venv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\daw74\PycharmProjects\EricaFinal\venv\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "C:\Users\daw74\PycharmProjects\EricaFinal\venv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\daw74\PycharmProjects\EricaFinal\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:/Users/daw74/PycharmProjects/EricaFinal/final.py", line 508, in image_processed_upload
    image_string = image_encoder(image_name)
  File "C:/Users/daw74/PycharmProjects/EricaFinal/final.py", line 161, in image_encoder
    with open(image, "rb") as image_file:
FileNotFoundError: [Errno 2] No such file or directory: ''
INFO:werkzeug:67.159.89.155 - - [13/Dec/2018 13:35:58] "POST /image/processed/upload HTTP/1.1" 500 -
INFO:werkzeug:67.159.89.155 - - [13/Dec/2018 13:35:58] "GET /image/processed/david.a.ward@duke.edu//HistogramEqualization HTTP/1.1" 404 -
```

**Comment by sputney13** [comment]
There's a strange quirk within the scroll-down (where you select which image you would like to process) that if you don't click on the image and just hit enter or "Upload Image", it doesn't actually save the image name and thus won't be able to process anything. That seems to be your issue, as the console should print off the name of image that you selected when you click it.

I am unsure how to write a test function that gives the same error as everything in client.py works when final.py is run locally. It's the viewing of the image in gui.py that throws the BytesError above (which we are currently trying to resolve).

**Comment by dward2** [comment]
Yes, that did change things.  Let me look a bit more now.

**Comment by EricaSkerrett** [comment]
Hi Dr. Ward, we think that's right! We had the server opening the image so we're transferring that function to the GUI. Then the GUI will just pass the base64 string. 
Thanks so much!

**Comment by EricaSkerrett** [comment]
(Also sorry the upload selection wasn't clear! It looks differently from macs than pcs so we're trying to fix that.)

**Comment by sputney13** [comment]
Fixed in #101
